### PR TITLE
DOC: update release notes with deprecation

### PR DESCRIPTION
**Pull request by j-bowhay**
#### What does this implement/fix?

Several deprecations were missing from the 1.13 release notes, this PR adds them
#### Additional information

cc @tylerjereddy @h-vetinari sorry not sure how I missed this, have been very busy with uni this year

### CODE REVIEW

File: doc/source/release/1.13.0-notes.rst [hunk 1]
```diff
@@ -140,12 +140,25 @@ Deprecated features
 - Complex dtypes in ``PchipInterpolator`` and ``Akima1DInterpolator`` have
   been deprecated and will raise an error in SciPy 1.15.0. If you are trying
   to use the real components of the passed array, use ``np.real`` on ``y``.
+- Non-integer values of `n` together with `exact=True` are deprecated for
```
Comment (j-bowhay) [inline doc/source/release/1.13.0-notes.rst:143]:
```suggestion
- Non-integer values of ``n`` together with ``exact=True`` are deprecated for
```

**Comment by lucascolley** [comment]
Do these changes belong under an "expired deprecations" section instead of "backwards-incompatible changes"?

Can we add a copy of https://github.com/scipy/scipy/blob/f43f724827c9e2d40a04b7ab4bd2d46a2632c241/doc/source/release/1.12.0-notes.rst?plain=1#L306-L309 and mention that although that change occurred in 1.12.0, it was missed from the release notes initially?

It also looks like you've picked up a submodule change.

**Comment by j-bowhay** [comment]
> Do these changes belong under an "expired deprecations" section instead of "backwards-incompatible changes"?

Oops yes...

> Can we add a copy of
> 
> https://github.com/scipy/scipy/blob/f43f724827c9e2d40a04b7ab4bd2d46a2632c241/doc/source/release/1.12.0-notes.rst?plain=1#L306-L309
> 
> and mention that although that change occurred in 1.12.0, it was missed from the release notes initially?

Sound like a good idea

> It also looks like you've picked up a submodule change.
 
agh yes too lazy with the `add .` I was actually having a bit of a battle with this
```
git submodule update --init                                                                                                                          ✔  scipy-dev  
fatal: transport 'file' not allowed
fatal: Fetched in submodule path 'scipy/_lib/pocketfft', but it did not contain 9367142748fcc9696a1c9e5a99b76ed9897c9daa. Direct fetching of that commit failed.
```

**Comment by lucascolley** [comment]
> Fetched in submodule path 'scipy/_lib/pocketfft', but it did not contain 9367142748fcc9696a1c9e5a99b76ed9897c9daa

Yeah I get stuff like this all the time, so annoying 😅 usually `cd`-ing to the submodule and fetching manually works for me

**Comment by j-bowhay** [comment]
wow I really have messed that up, think it might be easier to start with a new pr!

**Pull request closed without merging**